src/backend/iocp.zig

```diff
@@ -538,13 +538,58 @@ pub const Loop = struct {
             .close => |v| .{ .result = .{ .close = windows.CloseHandle(v.fd) } },
 
             .connect => |*v| action: {
-                const result = windows.ws2_32.connect(asSocket(v.socket), &v.addr.any, @as(i32, @intCast(v.addr.getOsSockLen())));
-                if (result != 0) {
-                    const err = windows.ws2_32.WSAGetLastError();
-                    break :action switch (err) {
-                        else => .{ .result = .{ .connect = windows.unexpectedWSAError(err) } },
+                const as_socket = asSocket(v.socket);
+                // Associate our socket with loop's completion port.
+                self.associate_fd(v.socket) catch unreachable;
```

Comment:
```suggestion
                self.associate_fd(completion.handle().?) catch unreachable;
```

Don't forget to update `Completion.handle` as well though